core/src/main/java/org/elasticsearch/search/aggregations/pipeline/movavg/models/LinearModel.java

```diff
@@ -41,6 +42,29 @@
     protected static final ParseField NAME_FIELD = new ParseField("linear");
 
     @Override
+    public <T extends Number> double[] predict(Collection<T> values, int numPredictions) {
```

Comment:
The two checks (`values.size == 0` and `numPredictions < 1`) seem to be done in all the models, maybe we should move them to the super class and have the sub-classes implement `doPredict()` which is called from the `predict()` method in the super class?